traits/tests/test_ui_notifiers.py

```diff
@@ -157,10 +157,11 @@ def test_notification_from_separate_thread_failure_case(self):
         # Given no registered ui handler
         self.enterContext(clear_ui_handler())
 
-        # When we set obj.foo to 3 on a separate thread.
+        # When we set obj.foo to 3 on a separate thread, and wait for
+        # that thread to complete.
         background_thread = threading.Thread(target=self.modify_obj)
         background_thread.start()
-        self.addCleanup(background_thread.join)
+        background_thread.join()
```

Comment:
We're invoking the same `self.addCleanup` command in `test_notification_from_separate_thread` (line 184) - is it appropriate for this test case or should it be changed also?

Reply:
I think it's fine for that test case, because we're explicitly waiting for the notification to arrive with an `await`.